### DISCUSSION TITLE
Is there a way to access the android activity if the host platform is android

### DISCUSSION THREAD

**Discussion by neerajshdev** [arkivanov/Decompose#748] [Q&A]
I have a WalletComponent in which there is an action 'startPurchase()' Its implementation uses Google Play billing that requires the Android activity context. 

Is there a similar way as my imaginary code below 
```
// inside the WalletCompoent.startPurchase()

val host = cmpContext.getHostPlatform()
if (host.platform is Activity) { 
// impl using acitvity
}

```

**Comment by arkivanov** [comment]
If you only support Android, then you can easily pass (or inject) Activity via component constructors. If your are using Kotlin Multiplatform, then you can probably leverage Inversion of Control (IoC): create an interface with platform-specific implementations, where the Android impl would depend on Activity.

All the above options assume that you are not using [retained components](https://arkivanov.github.io/Decompose/component/instance-retaining/#retained-components).

**Reply by neerajshdev** [reply]
That means I must create an Interface like **PlatformHostProvider** and then pass it to the component constructor.

**Reply by arkivanov** [reply]
I think you still need to perform purchases in a platform-specific way (your `// impl using acitvity` comment). Looks like you can extract an interface for purchases, or something similar. Maybe something along the lines:

```kotlin

// commonMain
interface PurchaseManager {
    fun purchase(...)
}

//androidMain
class AndroidPurchaseManager(activity: Activity) : PurchaseManager {
    override fun purchase(...) { ... }
}
```

And then pass `PurchaseManager` in your `WalletCompoent`. Just an idea.

**Reply by neerajshdev** [reply]
Creating an interface for PurcahseManager make sense, thanks.
